src/test/java/org/example/validator/ApplicationValidatorTest.java

```diff
@@ -0,0 +1,125 @@
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+
+        assertDoesNotThrow(() -> applicationValidator.validateApplication(applicationRequest));
+
+    }
+
+    @Test
+    public void validateApplication_shouldThrowDoesNotExistException_AuthServiceThrowsDoesNotExistException()
+    throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doThrow(DoesNotExistException.class).when(authService).checkEmailExists(applicationRequest.getEmail());
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+        assertThrows(DoesNotExistException.class, () ->
+                applicationValidator.validateApplication(applicationRequest));
+    }
+
+    @Test
+    public void validateApplication_shouldThrowSQLException_AuthServiceThrowsSQLException()
+            throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doThrow(SQLException.class).when(authService).checkEmailExists(applicationRequest.getEmail());
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+        assertThrows(SQLException.class, () ->
+                applicationValidator.validateApplication(applicationRequest));
+    }
+
+
+    @Test
+    public void validateApplication_shouldThrowDoesNotExistException_JobRoleServiceThrowsInvalidException()
+            throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doNothing().when(authService).checkEmailExists(applicationRequest.getEmail());
+        doThrow(DoesNotExistException.class).when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+        assertThrows(DoesNotExistException.class, () ->
+                applicationValidator.validateApplication(applicationRequest));
+    }
+
+    @Test
+    public void validateApplication_shouldThrowSQLException_JobRoleServiceThrowsSQLException()
+            throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
Same

Reply:
This has been corrected and committed. I sincerely apologise for creating the risk of Polar Bears working for Kainos.